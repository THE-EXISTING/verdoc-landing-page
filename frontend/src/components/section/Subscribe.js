import React, { useState } from 'react'
import styled from 'styled-components'
import media from 'styled-media-query'
import Button from '@material-ui/core/Button'
import TextField from '../common/TextField'
import R from '../resources/R'

const WraperContent = styled.div`
  margin-top: 2rem;
  width: 100%;
`
const WraperForm = styled.div`
  margin-top: 1rem;
  width: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
`
const TextOnBtnFront = styled.div`
  opacity: 0.6;
  font-size: 1em;
  color: ${R.colors.text};
  letter-spacing: 0.36px;
  margin-bottom: 0;
`
const TextBold = styled.div`
  display: inline;
  font-weight: bold;
`
const BtnSubscribe = styled(Button)`
  && {
    
    height: 56px;
    font-size: 1em;
    border-width: 0;
    display: inline-block;
    text-transform: lowercase;
    color: ${R.colors.white};
    background-color: ${R.colors.primary};

    &:hover {
      background-color: ${R.colors.primary} !important;
      -webkit-box-shadow: 0.75px 0.75px 5.5px 0.75px
        ${R.colors.border_btn_subsc_hover};
      -moz-box-shadow: 0.75px 0.75px 5.5px 0.75px
        ${R.colors.border_btn_subsc_hover};
      box-shadow: 0.75px 0.75px 5.5px 0.75px ${R.colors.border_btn_subsc_hover};
    }
  }
`
const Cutter = styled.div`
  display: inline;
  ${media.lessThan('medium')`
    display: block;
`}
`

const Subscribe = props => {
  const [email, setEmail] = useState('')

  const handleSubmit = event => {
    event.preventDefault()
    if (email !== '') {
      props.db
        .firestore()
        .collection('subscribers')
        .doc(email)
        .set({
          email: email,
          timestamp: new Date().getTime(),
          // timestamp: moment().format('x'),
        })
        .then(function() {
          console.log('Subscribe success', email)
        })
        .catch(function(error) {
          console.error('Error seting document: ', error)
        })
    }
  }
  const handleTyping = value => setEmail(value['value'])

  return (
    <WraperContent>
      <TextOnBtnFront>
        <Cutter>Please subscribe to get </Cutter>
        <Cutter>
          <TextBold>
            free 3 month usage.
            <span role="img" aria-label="money">
              ðŸ’°ðŸ’°ðŸ’°
            </span>
          </TextBold>
        </Cutter>
      </TextOnBtnFront>
      <WraperForm>
        <TextField handleTyping={handleTyping} />
        <BtnSubscribe
          variant="outlined"
          size="large"
          onClick={event => handleSubmit(event)}
        >
          subscribe
        </BtnSubscribe>
      </WraperForm>
    </WraperContent>
  )
}
export default Subscribe
