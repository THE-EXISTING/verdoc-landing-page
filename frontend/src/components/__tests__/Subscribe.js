import React from 'react'
import renderer from 'react-test-renderer'

import Subscribe from '../section/Subscribe'

describe('Subscribe', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Subscribe />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
