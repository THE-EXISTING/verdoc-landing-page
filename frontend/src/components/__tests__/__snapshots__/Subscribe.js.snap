// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Subscribe renders correctly 1`] = `
<div
  className="sc-htpNat gtHyGp"
>
  <div
    className="sc-ifAKCX kntUZg"
  >
    <div
      className="sc-gzVnrw Pltlk"
    >
      Please subscribe to get 
    </div>
    <div
      className="sc-gzVnrw Pltlk"
    >
      <span
        className="sc-EHOje fPRgQt"
      >
        free 3 month usage.
        <span
          aria-label="money"
          role="img"
        >
          ðŸ’°ðŸ’°ðŸ’°
        </span>
      </span>
    </div>
  </div>
  <div
    className="sc-bxivhb hOBMUX"
  >
    <div
      className="sc-bdVaJa iILNGh"
    >
      <div
        className="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root sc-bwzfXH fsszKJ"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
          data-shrink={false}
          htmlFor="standard-name"
        >
          Email
        </label>
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
          onClick={[Function]}
        >
          <fieldset
            aria-hidden={true}
            className="PrivateNotchedOutline-root-54 MuiOutlinedInput-notchedOutline"
            style={
              Object {
                "paddingLeft": 8,
              }
            }
          >
            <legend
              className="PrivateNotchedOutline-legend-55"
              style={
                Object {
                  "width": 0.01,
                }
              }
            >
              <span
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&#8203;",
                  }
                }
              />
            </legend>
          </fieldset>
          <input
            aria-invalid={false}
            autoComplete="email"
            className="MuiInputBase-input MuiOutlinedInput-input"
            disabled={false}
            id="standard-name"
            name="email"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="your@verdoc.io"
            required={false}
            type="email"
            value=""
          />
        </div>
        
      </div>
    </div>
    <button
      className="MuiButtonBase-root MuiButton-root sc-bZQynM ijCzVY MuiButton-outlined MuiButton-sizeLarge"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        subscribe
      </span>
    </button>
  </div>
</div>
`;
